src/createStore.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-console */
+import _ from 'lodash';
+import { BehaviorSubject } from 'rxjs';
+
+class BaseStore {
+  constructor(options = {}) {
+    this.options = {
+      initialState: null,
+      thunkArgument: null,
+      appendAction: false,
+      reducer: state => state,
+      enableLogger: true,
+      console: console,
+      ...options,
+    };
+
+    this.internalState$ = new BehaviorSubject(this.options.initialState)
+      .scan((previousState, action) => {
+        let updatedState;
+        const d = new Date();
+        const prettyDate = `${d.getHours()}:${d.getMinutes()}:${d.getSeconds()}.${d.getMilliseconds()}`;
+
+        try {
+          updatedState = this.options.reducer(previousState, action);
+        } catch (error) {
+          if (action && action.type) {
+            this.options.console.error(`Error processing @ ${prettyDate} ${action.type}:`);
+          }
+          this.options.console.error(error);
+
+          return previousState;
+        }
+
+        // logger in non-production mode only
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
@mAiNiNfEcTiOn: I kept it like it was working before. The main reason is because of our production bundles.

When `NODE_ENV === production`, the full code-block inside this `if` would not end up in the bundle processed by Webpack (in our case, it is `vendors.js` file).

If we want to check it in runtime instead, it means the logger specific code would have to be in the bundle all the time. Do we want that in this PR?

Please correct me if you understand your comment wrong.